COVIDWatch iOS/ViewControllers/Home.swift

```diff
@@ -17,6 +17,45 @@ class Home: BaseViewController {
     var testedButton = Button(text: "Tested for COVID-19?",
                               subtext: "Share your result anonymously to help keep your community stay safe.")
     var infoBanner = InfoBanner(text: "You may have been in contact with COVID-19")
+
+    var observer: NSObjectProtocol?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // prevent removal of permissions by accident after allowing them
+        self.observer = NotificationCenter.default.addObserver(
+            forName: UIApplication.didBecomeActiveNotification,
+            object: nil, queue: OperationQueue.main
+        ) { [weak self] _ in
+            self?.checkBluetoothPermission()
+        }
+    }
+
+    func checkBluetoothPermission() {
+        BluetoothPermission.sharedInstance.checkPermission({
+            // nothing to do
+            print("Bluetooth is still enabled")
+        }, {
+            // can also change the home page to visually inform the user that bluetooth is not enabled
+            let bluetoothSettingsAlert = UIAlertController(
+                title: NSLocalizedString("Bluetooth Required", comment: ""),
+                message: "Please turn on Bluetooth in Settings", preferredStyle: .alert
+            )
+            bluetoothSettingsAlert.addAction(
+                UIAlertAction(
+                    title: NSLocalizedString("Open Settings", comment: ""),
+                    style: .default,
+                    handler: { _ in
+                        if let url = URL(string: UIApplication.openSettingsURLString) {
+                            UIApplication.shared.open(url)
+                        }
+                    }
+                )
+            )
+            self.present(bluetoothSettingsAlert, animated: true)
```

Comment:
Should this closure use a weak self reference?  I think that since you're saving the closures as properties inside the BluetothPermissions class, they will hold a strong reference to this class (leak) until the closures are nil'ed out.  If you make this weak, I think it should all be fine.

Reply:
👍 Fixed!